EDU-1723: Rewords coverage of custom Activity Type naming (#2960)
temporalio/documentation: docs/develop/java/core-application.mdx

**How to customize your Activity Type using the Java SDK.**

Each Activity has a Type called the Activity 'name'.
This name appears in the Workflow Execution Event History Summary, listed for each Activity Task.
The name lets you identify Activity Types called during the Execution.

Custom Activity Type names prevent name collisions across interfaces and Workflows.
They offer descriptive Activity method names without concerns about re-using those names elsewhere in your project.
They also support code management, especially in larger projects with many Activities.
For example, you might use a prefix to group related activities together.
Custom names also distinguish keys for gathering metrics without name conflicts.

The following examples show how to set custom names for your Activity Type.

**Default behavior**

By default, an Activity Type is the method name with the first letter capitalized:

```java
@ActivityInterface
public interface GreetingActivities {
    String sendMessage(String input);

>     @ActivityMethod

Reviewer: I recall getting feedback that we don't need to decorate the methods with `@ActivityMethod` _unless_ we are doing something like changing the name. It doesn't hurt anything, and personally I like having it but I've been convinced it's not necessary. All methods within an Activity interface are considered Activities.

Reviewer: Ah, as I read further I see the point you were making. Lovely. Ignore the above.

Author: Should I mention that `sendMessage` explicitly inherits prefixes and is an `@ActivityMethod`? I think that might be a good thing.

Author: It's mentioned a lot already, so I added examples for that as well, to make it clear and punch it home.

    String composeGreeting(String greeting, String language);
}
```

* Method Name: `composeGreeting`
* Outcome: `ComposeGreeting `

**Custom Prefix**

> Using the `namePrefix` parameter in the `@ActivityInterface` annotation adds a prefix to each Activity Type name mentioned in the interface, unless the prefix is specifically overridden:

Reviewer: I did not know this. Neat.

**How to customize your Activity Type using the Java SDK.**

> Each Activity has a Type called the Activity 'name'.

Author: Love this edit.

> This name appears in the Workflow Execution Event History Summary, listed for each Activity Task.

Author: I want the reader to know what to look for and Summary isn't open by default.  I'm going to re-word using "tab" to be more specific.

> This name appears in the Workflow Execution Event History Summary, listed for each Activity Task.

Author: > This name appears in the Workflow Execution Event History in the Summary tab for each Activity Task.

> This name appears in the Workflow Execution Event History Summary, listed for each Activity Task.

Author: Reworded.

The name lets you identify Activity Types called during the Execution.

Custom Activity Type names prevent name collisions across interfaces and Workflows.
They offer descriptive Activity method names without concerns about re-using those names elsewhere in your project.
They also support code management, especially in larger projects with many Activities.
For example, you might use a prefix to group related activities together.

> Custom names also distinguish keys for gathering metrics without name conflicts.

Reviewer 2: I can't verify the accuracy of this. I'm not saying that it's wrong, just that I don't know enough about metrics to confirm that it's right (and don't have time to research it). I'd recommend either verifying this with an engineer or omitting the statement.

Author: Metrics info was pulled right from the code doc for the SDK.

The following examples show how to set custom names for your Activity Type.

**Default behavior**

By default, an Activity Type is the method name with the first letter capitalized:

```java
@ActivityInterface
public interface GreetingActivities {
    String sendMessage(String input);

    @ActivityMethod
    String composeGreeting(String greeting, String language);
}
```

- Method Name: `sendMessage`

> - Outcome: `SendMessage`

Reviewer 2: ```suggestion
- Activity Type: `SendMessage`
```

I think "Activity Type" would be more clear than "Outcome" here.

Author: Love it. Changing it globally.

- Method Name: `composeGreeting`

> - Outcome: `ComposeGreeting`

Reviewer 2: ```suggestion
- Activity Type: `ComposeGreeting`
```

I think "Activity Type" would be more clear than "Outcome" here.

**Custom Prefix**

Using the `namePrefix` parameter in the `@ActivityInterface` annotation adds a prefix to each Activity Type name mentioned in the interface, unless the prefix is specifically overridden:

```java
@ActivityInterface(namePrefix = "Messaging_")
public interface GreetingActivities {
    String sendMessage(String input);

    @ActivityMethod
    String composeGreeting(String greeting, String language);
}
```

- Method Name: `sendMessage`

> - Outcome: `Messaging_SendMessage`

Reviewer 2: ```suggestion
- Activity Type: `Messaging_SendMessage`
```

- Method Name: `composeGreeting`

> - Outcome: `Messaging_ComposeGreeting`

Reviewer 2: ```suggestion
- Activity Type: `Messaging_ComposeGreeting`
```

> The Activity Type method name is capitalized, even when using a prefix.

Reviewer 2: ```suggestion
The Activity Type is capitalized, even when using a prefix.
```

There is a method name and there is an Activity Type (which is based on the method name). but there's no such thing as an Activity Type method name AFAIK.

**Custom Name**

To override the default name and any inherited prefixes, use the `name` parameter in the `@ActivityMethod` annotation:

```java
@ActivityInterface(namePrefix = "Messaging_")
public interface GreetingActivities {
    String sendMessage(String input);

    @ActivityMethod
    String composeGreeting(String greeting, String language);
    @ActivityMethod(name = "farewell")
    String composeFarewell(String farewell, String language);
}
```

Using the `name` parameter won't automatically capitalize the result:

- Method Name: `sendMessage`

> - Outcome: `Messaging_SendMessage`

Reviewer 2: ```suggestion
- Activity Type: `Messaging_SendMessage`
```

- Method Name: `composeGreeting`

> - Outcome: `Messaging_ComposeGreeting`

Reviewer 2: ```suggestion
- Activity Type: `Messaging_ComposeGreeting`
```

- Method Name: `composeFarewell`

> - Outcome: `farewell `

Reviewer 2: ```suggestion
- Activity Type: `farewell `
```

Be cautious with names that contain special characters, as these can be used as metric tags.

> Systems like Prometheus may ignore metrics with tags using unsupported characters.

Reviewer 2: ```suggestion
Systems such as Prometheus may ignore metrics with tags using unsupported characters.
```

Recommend using "such as" instead of "like" here, since you're giving a specific example instead of making an analogy.

Author: Adopted as suggested.
